.github/workflows/release.yml

```diff
@@ -0,0 +1,173 @@
+        uses: softprops/action-gh-release@17cd0d34deddf848fc0e7d9be5202c148c270a0a
+        with:
+          name: ${{ github.ref_name }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          body_path: ./tmp/checksums.txt
+          files: |
+            ./tmp/release.yml
+            ./tmp/package.yml
+            ./tmp/metadata.yml
+          draft: true
+          prerelease: true
+      - name: Get uploaded release YAML checksum
+        uses: actions/github-script@v6
+        id: get-checksums-from-draft-release
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          result-encoding: string
+          script: |
+            var crypto = require('crypto');
+            const { owner, repo } = context.repo;
+            // https://docs.github.com/en/rest/reference/repos#list-releases
+            // https://octokit.github.io/rest.js/v18#repos-list-releases
+            var releases = await github.rest.repos.listReleases({
+              owner: owner,
+              repo: repo
+            });
+            var crypto = require('crypto')
+            var fs = require('fs')
+            const url = require('url');
+            const https = require('https');
+            checksums = {}
+            for (const r of releases["data"]) {
+              if (r.draft && `refs/tags/${r.tag_name}` == "${{ github.ref }}") {
+                for (const asset of r.assets) {
+                  var release_asset = await github.rest.repos.getReleaseAsset({ headers: {accept: `application/octet-stream`}, accept: `application/octet-stream`, owner: owner, repo: repo, asset_id: asset.id });
+                  const hash = crypto.createHash('sha256');
+                  let http_promise = new Promise((resolve, reject) => {
+                    https.get(release_asset.url, (stream) => {
+                      stream.on('data', function (data) {
+                        hash.update(data);
+                      });
+                      stream.on('end', function () {
+                        checksums[asset.name]= hash.digest('hex');
+                        resolve(`${asset.name}`);
+                      });
+                    });
+                  });
+                  await http_promise;
+                }
+              }
+            }
+            console.log(checksums)
+            return `${checksums['release.yml']}  ./tmp/release.yml
+            ${checksums['package.yml']}  ./tmp/package.yml
+            ${checksums['metadata.yml']}  ./tmp/metadata.yml`
+      - name: Verify uploaded artifacts
+        if: startsWith(github.ref, 'refs/tags/')
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: |
+          set -e -x
+          cat ./tmp/checksums.txt
+          diff ./tmp/checksums.txt <(cat <<EOF
+          ${{steps.get-checksums-from-draft-release.outputs.result}}
+          EOF
+          )
+  validate-package-cr:
```

Comment:
Would it make sense to run the e2e tests before creating the draft release as a separate step? This would remove some redundant steps like checkout, setup Go, installing carvel tools and setting up minikube and a draft release wouldn't be created if the e2e tests are failing. 
(Although I guess the the reason for having a separate job might be to validate the draft release, but it also seems like testing the action that creates the draft release)

Reply:
Right, this is the "hey we successfully built a release, and it's in draft - we ought to validate that the PackageCR that was generated and the live imgpkg bundle works"

I am not really sure if adding this beforehand adds much, other than maybe just not having a draft release created, but it's so easily deletable / overridable. Didn't feel worth it.

But maybe I am missing a concern of yours

Reply:
I don't really have any strong opinions, I just thought that having a separate step in the same job would be beneficial in 2 ways. 
- Draft release is not created if there's an issue with the package.yml generated.
- Having a single job would eliminate a few redundant steps which would slightly decrease the workflow run time.

But just like you have mentioned, these are not very significant advantages, so we can go ahead with two separate jobs :)

Reply:
ah i see. Yanno maybe it's fine to merge em, while I was testing it was nice to only re-run the failed job. I am starting to lean to the append this testing to the end of the other job.